packages/network_info_plus/network_info_plus/example/lib/main.dart

```diff
@@ -97,18 +98,13 @@ class _MyHomePageState extends State<MyHomePage> {
         wifiSubmask;
 
     try {
-      if (!kIsWeb && Platform.isIOS) {
-        // ignore: deprecated_member_use
-        var status = await _networkInfo.getLocationServiceAuthorization();
-        if (status == LocationAuthorizationStatus.notDetermined) {
-          // ignore: deprecated_member_use
-          status = await _networkInfo.requestLocationServiceAuthorization();
-        }
-        if (status == LocationAuthorizationStatus.authorizedAlways ||
-            status == LocationAuthorizationStatus.authorizedWhenInUse) {
+      if (!kIsWeb && (Platform.isAndroid || Platform.isIOS)) {
+        // Request permissions as recommended by the plugin documentation:
+        // https://github.com/fluttercommunity/plus_plugins/tree/main/packages/network_info_plus/network_info_plus
+        if (await Permission.locationWhenInUse.request().isGranted) {
```

Comment:
"location when in use" should be enough for this example. If a user wants to use "location always" the code gets a bit more complicated, according to the permission_handler plugin devs. (we were probably doing that wrong in the plugin)